pkg/chunk/aws/storage_client.go

```diff
@@ -251,9 +257,13 @@ func (a storageClient) BatchWrite(ctx context.Context, input chunk.WriteBatch) e
 				continue
 			} else if ok && awsErr.Code() == validationException {
 				// this write will never work, so the only option is to drop the offending items and continue.
-				// TODO: add more debug options for capturing data/telemetry about the offending items?
 				level.Warn(util.Logger).Log("Data lost while flushing to Dynamo: %v", awsErr)
 				level.Debug(util.Logger).Log("Dropped request details: \n%v", requests)
+				// recording the drop counter separately from recordDynamoError(), as the error code alone may not provide enough context
+				// to determine if a request was dropped (or not)
+				for tableName := range requests {
+					dynamoDroppedRequests.WithLabelValues(tableName, validationException, "DynamoDB.BatchWriteItem")
```

Comment:
Need to actually increment the counter after specifying the labels.

Reply:
🤦‍♂️  ... well this would be why I couldn't find the counter incrementing in my test data this morning. ugh